### ISSUE TITLE
我按照100个batch进行一次eval_model，发现和原来的结果不同

### ISSUE THREAD

**Issue by programmeguru** [shibing624/text2vec#94]
all.jsonl在训练中会有1360个batch，我尝试每100个batch在自身进行一次eval_model，仅增加了一行判断的代码，但是和按轮次保存的结果不一致，包括loss曲线和最终结果，请问这是什么原因呢，按道理eval_model设置了model.eval()以及with torch.no_grad()应该是不存在影响训练的梯度改变

**Comment by shibing624** [comment]
do you set seed the same?

**Comment by shibing624** [comment]
最终效果应该是差不多的吧

**Comment by programmeguru** [comment]
差2个点，但是我不能理解为什么会不同，因为我根本没有改变训练过程，只不过从每一轮进行一次eval_model改为在轮次中间多次eval_model会造成loss曲线不一致，看起来是验证过程也进行了训练一样

**Comment by programmeguru** [comment]
在同样的step时，两种策略的loss都是不一致的
